src/Compilers/Core/Portable/InternalUtilities/TextChangeRangeExtensions.cs

```diff
@@ -94,5 +98,339 @@ internal static class TextChangeRangeExtensions
 
             return new TextChangeRange(TextSpan.FromBounds(currentStart, currentOldEnd), currentNewEnd - currentStart);
         }
+
+        public static TextChangeRange ToTextChangeRange(this TextChange textChange)
+        {
+            return new TextChangeRange(textChange.Span, textChange.NewText?.Length ?? 0);
+        }
+
+        /// <summary>
+        /// Merges the new change ranges into the old change ranges, adjusting the new ranges to be with respect to the original text
+        /// (with neither old or new changes applied) instead of with respect to the original text after "old changes" are applied.
+        ///
+        /// This may require splitting, concatenation, etc. of individual change ranges.
+        /// </summary>
+        /// <remarks>
+        /// Both `oldChanges` and `newChanges` must contain non-overlapping spans in ascending order.
+        /// </remarks>
+        public static ImmutableArray<TextChangeRange> Merge(ImmutableArray<TextChangeRange> oldChanges, ImmutableArray<TextChangeRange> newChanges)
```

Comment:
Consider moving back to the original location and just changing the accessibility.

Reply:
There are not any IVTs to Features.CSharp from MS.CA, and adding IVTs causes tons of build errors as now it thinks there are duplicate types (e.g. ArrayBuilder, DeclarationInfo, nullable attributes).